clang/test/CodeGenSYCL/sycl-framework-metadata.cpp

```diff
@@ -0,0 +1,46 @@
+// Test checks that !sycl-framework metadata is emitted only to functions
+// whose topmost namespace is sycl.
+
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -emit-llvm %s -o - | FileCheck %s
+
+// Check that all functions on sycl::* namespace are marked with !sycl-framework metadata
+namespace sycl {
+  // CHECK: @_ZN4sycl4bar1Ev() {{.*}} !sycl-framework
+  void bar1() {}
+
+  namespace V1 {
+    // CHECK: @_ZN4sycl2V14bar2Ev() {{.*}} !sycl-framework
+    void bar2() {}
+  }
+}
+
+// Check that V1::sycl::* functions are not marked with !sycl-framework metadata
+// since topmost namespace is V1 instead of sycl.
+namespace V1 {
+  namespace sycl {
+    // CHECK-NOT: @_ZN2V14sycl4bar3Ev() {{.*}} !sycl-framework
+    void bar3() {}
+  }
+}
+
```

Comment:
How about something like:

```
namespace {
  namespace sycl {
    void bar4() {}
  }
}
```

Reply:
This case is good.

There is a problem that in that case `sycl` namespace becomes ambiguous with the first namespace in the test. I will mitigate it with a separate test file.